mautic/mautic
# [BUG] Incorrect redirect when trying to edit a Segment that is checked out by another user

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N/A
| Automated tests included? | N/A
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N/A
| Deprecations? | N/A

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

When attempting to edit a segment that is checked out by another user, you get redirected to the Detail view of the Contact with the same ID as the segment you tried to edit. All this happens even though the URL in the location bar is correct.

Scenario: Segment with id 1 is checked out by John Doe (user id 1). You (user id 2) click edit. You're redirected to the Details page for Contact with id 1 and given the error message to override the lock.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Have a mautic instance with multiple users. Open 2 separate browsers and log in to that instance with a different user in each browser.
2. Create a segment. Ensure that you have a contact that shares an id with that segment. (In a fresh installation, just create a segment and a contact, and they should both have an id of 1).
3. In one browser, edit the segment.
4. In the other browser, go to the details page of the segment (click the segment name in the list view), then click the "Edit" button in the top right.
5. You'll see that you're redirected to the Details page for the contact with the same id as that segment.

#### Steps to test this PR:
1. Apply PR
2. Retest
3. See that in step 5, instead of being redirected to the contact detail page, you stay on the segment detail page and that the flash message is correctly shown.
4. Ensure general navigation and redirects work with creating, saving, deleting, etc.
